src/primitives/certificate.cpp

```diff
@@ -262,6 +295,31 @@ CAmount CScCertificate::GetValueOfChange() const
     return nValueOut;
 }
 
+libzendoomc::ScFieldElement calculateCertDataHash(const CScCertificate& cert) {
+    // TODO Replace with hash of merkel tree supplied by SC.
+    libzendoomc::ScFieldElement certDataHash;
+    certDataHash.SetHex(cert.GetHash().GetHex());
+    return certDataHash;
+}
+
+libzendoomc::ScFieldElement calculateCumulativeCertDataHash(const libzendoomc::ScFieldElement& prevCumulativeHash, const libzendoomc::ScFieldElement& prevCertHash) {
+    auto prevCumulativeHashInput = zendoo_deserialize_field(prevCumulativeHash.begin());
```

Comment:
same as in `CScWCertProofVerification::verifyScCert`we need to check `prevCumulativeHashInput` for `nullptr`